spaceship/lib/spaceship/base.rb

```diff
@@ -59,6 +59,9 @@ def each(&block)
       def to_json(*a)
         h = @hash.dup
         h.delete(:application)
+        if h.key?(:version) && h[:version].kind_of?(Spaceship::Tunes::AppVersion)
```

Comment:
I had the same thoughts as you, and initially I put this logic into the complete! method, however deleting the version variable there could be a bit confusing as well, because then after calling the complete method I am not able to access the version variable. There are a lot of ways this could be implemented, I chose the easiest/clearest one. (not the prettiest, I admit) I am open to suggestions though, which method would you move this logic to? 

Do you know any other classes that is a subclass of this base class and needs to send the version variable to itunes connect? If not, then we could just simply delete the version variable alongside with the application variable.